enhancements/ingress/lb-subnet-selection-aws.md

```diff
@@ -0,0 +1,571 @@
+automatic removal of the service annotation on upgrade.
+
+**Note**: Cluster admins upgrading with an unmanaged subnet annotation don't
+need to delete the service to propagate the subnet values, instead they can follow the
+[Unmanaged Subnet Annotation Migration Workflow](#unmanaged-subnet-annotation-migration-workflow).
+
+##### Reason #3: The CCM Doesn't Reconcile NLB Subnets Updates after creation.
+
+The CCM doesn't reconcile `service.beta.kubernetes.io/aws-load-balancer-subnets`
+annotation updates to NLBs once it has been created. However, requiring a service
+deletion will effectuate these changes.
+
+See [CCM Doesn't Reconcile NLB Subnets Updates](#ccm-doesnt-reconcile-nlb-subnets-updates)
+for more details on why the CCM doesn't reconcile these updates.
+
+### Implementation Details/Notes/Constraints
+
+When an IngressController is created with
+`spec.endpointPublishingStrategy.loadBalancer.providerParameters.aws.subnets`
+specified, the Ingress Operator will set the
+`service.beta.kubernetes.io/aws-load-balancer-subnets` annotation
+on the LoadBalancer-type Service.
+
+However, if `spec.endpointPublishingStrategy.loadBalancer.providerParameters.aws.subnets`
+is updated on an existing IngressController, the Ingress Operator will NOT
+update `service.beta.kubernetes.io/aws-load-balancer-subnets` until the
+cluster admin deletes the LoadBalancer-type service as instructed by the
+`LoadBalancerProgressing` status (see [Effectuating Subnet Updates](#effectuating-subnet-updates)).
+When the Ingress Operator recreates the LoadBalancer-type service, it will
+then configure `service.beta.kubernetes.io/aws-load-balancer-subnets` with
+the new `Subnets` value.
+
+The `status.endpointPublishingStrategy.loadBalancer.providerParameters.aws.subnets`
+will eventually reflect the configured subnet value by mirroring the value of
+`service.beta.kubernetes.io/aws-load-balancer-subnets` on the service. There are
+two scenarios in which the `Subnets` status won't be equal to the `Subnets` spec:
+
+1. The cluster admin manually configured the `service.beta.kubernetes.io/aws-load-balancer-subnets`
+   annotation on the service.
+2. The cluster admin configured the IngressController's `Subnets` spec, but hasn't
+   effectuated the change by deleting the service (see [Effectuating Subnet Updates](#effectuating-subnet-updates)).
+
+Note that updating the status with admin instructions is consistent with the approach in [LoadBalancer Allowed Source Ranges](/enhancements/ingress/lb-allowed-source-ranges.md),
+but diverges with [Ingress Mutable Publishing Scope](/enhancements/ingress/mutable-publishing-scope.md).
+The mutable publishing scope design set `status.endpointPublishingStrategy.loadBalancer.scope`
+to the initial value of `spec.endpointPublishingStrategy.loadBalancer.scope` so that it could
+detect when the spec for `Scope` changed. Therefore, `status.endpointPublishingStrategy.loadBalancer.scope`
```

Comment:
This doesn't explain how this approach allows "detection of the change".  You'd have to explain that, or omit the sentence because it doesn't add anything.  I prefer omitting it.